app/scripts/controllers/paymentModal.js

```diff
@@ -53,20 +60,39 @@ angular.module('confRegistrationWebApp')
       }
 
       $scope.processing = true;
-      var path = 'payments?sendEmailReceipt=' + $scope.newTransaction.sendEmailReceipt;
-      delete $scope.newTransaction.sendEmailReceipt;
-      if($scope.newTransaction.paymentType === 'ADDITIONAL_EXPENSE') {
+      var transaction = angular.copy($scope.newTransaction);
+      var path = 'payments?sendEmailReceipt=' + transaction.sendEmailReceipt;
+      delete transaction.sendEmailReceipt;
+
+      if(transaction.paymentType === 'ADDITIONAL_EXPENSE') {
         path = 'expenses';
-        delete $scope.newTransaction.paymentType;
+        delete transaction.paymentType;
       } else {
-        $scope.newTransaction.readyToProcess = true;
+        transaction.readyToProcess = true;
+      }
+
+      if(transaction.paymentType === 'SCHOLARSHIP') {
+        transaction.scholarship.scholarshipStatus = 'APPROVED';
       }
 
-      if($scope.newTransaction.paymentType === 'SCHOLARSHIP') {
-        $scope.newTransaction.scholarship.scholarshipStatus = 'APPROVED';
+      if(transaction.paymentType === 'CREDIT_CARD'){
+        $http.get('payments/ccp-client-encryption-key').success(function(ccpClientEncryptionKey) {
+          ccp.initialize(ccpClientEncryptionKey);
```

Comment:
Not a huge deal, but you could initialize ccp earlier, before the transaction processing work happens. This works, but it will probably cause a longer lag in the user experience than is necessary.